cert/aziot-certd/src/lib.rs

```diff
@@ -130,6 +146,24 @@ impl Api {
             Err(err) => Err(Error::Internal(InternalError::DeleteFile(err))),
         }
     }
+
+    fn authorize(&self, user: libc::uid_t, id: &str) -> bool {
+        // Root user is always authorized.
+        if user == 0 {
+            return true;
+        }
+
+        // Authorize user based on stored principals config.
+        if let Some(certs) = self.principals.get(&user) {
+            for cert in certs {
```

Comment:
(After making the change to `principals_to_map` from my comment there.)

`return certs.any(|cert| cert.is_match(id));`

Reply:
Fixed